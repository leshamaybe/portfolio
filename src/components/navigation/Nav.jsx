import React from 'react';
import Link from 'next/link';
import s from './nav.module.scss';
import cn from 'classnames';
import { useRouter } from 'next/router';

export const navItems = [
    {
        href: '/works',
        title: 'Works',
    },
    {
        href: '/about',
        title: 'About',
    },
    {
        href: '/resume',
        title: 'Resume',
    },
    {
        href: '/contact',
        title: 'Contact',
    },
];

const Nav = () => {
    const { pathname } = useRouter();

    return (
        <nav className={s.navbar}>
            <ul>
                {navItems.map(({ href, title }) => (
                    <li key={title}>
                        <Link
                            href={href}
                            className={cn(s.link, { [s.current]: href === pathname })}>
                            {title}
                        </Link>
                    </li>
                ))}
            </ul>
        </nav>
    );
};

export default Nav;
