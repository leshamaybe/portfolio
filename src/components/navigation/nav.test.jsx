import Nav from './Nav';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Nav', () => {
    it('Nav renders correctly', () => {
        const { container } = render(<Nav />);

        expect(container).toMatchSnapshot();
    });
});
