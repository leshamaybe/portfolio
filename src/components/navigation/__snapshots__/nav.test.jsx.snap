// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Nav renders correctly 1`] = `
<div>
  <nav
    class="navbar"
  >
    <ul>
      <li>
        <a
          class="link"
          data-testid="About"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          class="link"
          data-testid="Resume"
          href="/resume"
        >
          Resume
        </a>
      </li>
      <li>
        <a
          class="link"
          data-testid="Contact"
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
